### ISSUE TITLE
RecursionError when importing from SciPy

### ISSUE THREAD

**Issue by the-rccg** [thebjorn/pydeps#51]
EDIT: 
Error is caused from an import of scipy.interpolate and was misattributed to a self import previously.

Original:

It appears that a RecursionError is invoked when a module imports itself. max-bacon and cluster limits dont seem to limit the recursion level and no other sanity check seems to be performed if the module is already in the hierarchy, meaning the maximum recursion depth is always exceeded. Specifically, the chain I encountered is the following: 

```
  File "...\anaconda3\lib\site-packages\pydeps\py2depgraph.py", line 116, in import_hook
    return mf27.ModuleFinder.import_hook(self, name, caller, fromlist, level)
  File "...\anaconda3\lib\site-packages\pydeps\mf27.py", line 134, in import_hook
    m = self.load_tail(q, tail)
  File "...\anaconda3\lib\site-packages\pydeps\mf27.py", line 213, in load_tail
    m = self.import_module(head, mname, m)
  File "...\anaconda3\lib\site-packages\pydeps\py2depgraph.py", line 138, in import_module
    module = mf27.ModuleFinder.import_module(self, partnam, fqname, parent)
  File "...\anaconda3\lib\site-packages\pydeps\mf27.py", line 283, in import_module
    m = self.load_module(fqname, fp, pathname, stuff)
  File "...\anaconda3\lib\site-packages\pydeps\py2depgraph.py", line 146, in load_module
    self, fqname, fp, pathname, (suffix, mode, kind)
  File "\anaconda3\lib\site-packages\pydeps\mf27.py", line 335, in load_module
    self.scan_code(co, module)
  File "...\anaconda3\lib\site-packages\pydeps\mf27.py", line 511, in scan_code
    self.scan_code(literal, module)
  File "...\anaconda3\lib\site-packages\pydeps\mf27.py", line 511, in scan_code
    self.scan_code(literal, module)
  File "...\anaconda3\lib\site-packages\pydeps\mf27.py", line 472, in scan_code
    self._safe_import_hook(name, module, fromlist, level=level)
  File "...\anaconda3\lib\site-packages\pydeps\mf27.py", line 353, in _safe_import_hook
    self.import_hook(name, caller, level=level)
  File "...\anaconda3\lib\site-packages\pydeps\py2depgraph.py", line 116, in import_hook
    return mf27.ModuleFinder.import_hook(self, name, caller, fromlist, level)
  File "...\anaconda3\lib\site-packages\pydeps\mf27.py", line 130, in import_hook
    q, tail = self.find_head_package(parent, name)
  File "...\anaconda3\lib\site-packages\pydeps\mf27.py", line 191, in find_head_package
    q = self.import_module(head, qname, parent)
  File "...\anaconda3\lib\site-packages\pydeps\py2depgraph.py", line 139, in import_module
    self._add_import(module)
  File "...\anaconda3\lib\site-packages\pydeps\py2depgraph.py", line 129, in _add_import
    rpath = os.path.split(module.__file__)[0].lower()
  File "...\anaconda3\lib\ntpath.py", line 184, in split
    seps = _get_bothseps(p)
  File "...\anaconda3\lib\ntpath.py", line 35, in _get_bothseps
    if isinstance(path, bytes):
RecursionError: maximum recursion depth exceeded while calling a Python object
```

**Comment by thebjorn** [comment]
I'll need some more information since a simple self-import:
```
(dev) go|c:\srv\tmp\tstselfimport> cat a.py
import a
```
is handled without problems:
```
(dev) go|c:\srv\tmp\tstselfimport> pydeps a.py
```
giving:

![a](https://user-images.githubusercontent.com/1193143/75365268-418c5300-58bd-11ea-8022-7de21867dbed.png)

**Comment by the-rccg** [comment]
You're right. After more testing I've found it was a nested Scipy import that caused the crash. Specifically, some modules of scipy cause an error.

Minimal example
`from scipy.interpolate import *`

results in maximum RecursionError

**Comment by thebjorn** [comment]
Fixed in version 1.9.8.
